include/zephyr/bluetooth/services/cts.h

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2024 Croxel Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_CTS_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_SERVICES_CTS_H_
+
+/**
+ * @brief Current Time Service (CTS)
+ * @defgroup bt_cts Current Time Service (CTS)
+ * @ingroup bluetooth
+ * @{
+ *
+ */
+
+#include <stdint.h>
+#include <zephyr/posix/time.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief CTS time update reason bits as defined in the specification
+ */
+enum bt_cts_update_reason {
+	/* Unknown reason of update no bit is set */
+	BT_CTS_UPDATE_REASON_UNKNOWN = 0,
+	/* When time is changed manually e.g. through UI */
+	BT_CTS_UPDATE_REASON_MANUAL = BIT(0),
+	/* If time is changed through external reference */
+	BT_CTS_UPDATE_REASON_EXTERNAL_REF = BIT(1),
+	/* time changed due to timezone adjust */
+	BT_CTS_UPDATE_REASON_TIME_ZONE_CHANGE = BIT(2),
+	/* time changed due to dst offset change */
+	BT_CTS_UPDATE_REASON_DAYLIGHT_SAVING = BIT(3),
+};
+
+/**
+ * @brief Current Time service data format, Please refer to
+ * specifications for more details
+ */
+struct bt_cts_time_format {
+	uint16_t year;
+	uint8_t mon;
+	uint8_t mday;
+	uint8_t hours;
+	uint8_t min;
+	uint8_t sec;
+	uint8_t wday;
+	uint8_t fractions256;
+	uint8_t reason;
+} __packed;
+
+/** @brief Current Time Service callback structure */
+struct bt_cts_cb {
+	/** @brief Current Time Service notifications changed
+	 *
+	 * @param enabled Flag that is true if notifications were enabled, false
+	 *                if they were disabled.
+	 */
+	void (*notification_changed)(bool enabled);
+
+	/**
+	 * @brief Current Time is updated using Gatt Attr write, It is the
+	 * responsibility of the application to save this information locally
+	 * and use it for future usage
+	 *
+	 * @param cts_time [IN] updated time information received
+	 *
+	 * @return 0 application has decoded it successfully
+	 *           negative error codes on failure
+	 *
+	 */
+	int (*cts_time_write)(struct bt_cts_time_format *cts_time);
+
+	/**
+	 * @brief When current time Read request or notification is triggered, CTS uses
+	 * this callback to retrieve current time information from application. Application
+	 * must implement it and provide cts formatted current time information
+	 *
+	 * @note this is mandatory callback api and should be implemented by the application
+	 *
+	 * @param cts_time [IN]
+	 *
+	 * @return 0 application has encode it successfully
+	 *           negative error codes on failure
+	 */
+	int (*fill_current_cts_time)(struct bt_cts_time_format *cts_time);
+};
+
+/**
+ * @brief This API should be called at boot to ensure proper functioning of CTS service
+ *
+ * @param cb pointer to required callback
+ * @return int   0 on success
+ *               negative error codes on failure
+ */
+int bt_cts_init(const struct bt_cts_cb *cb);
+
+/**
+ * @brief Update CTS Server time and notify all clients
```

Comment:
You're not updating the time anymore since CTS doesn't have a backend.
You can rephrase to only "Notify all connected clients that have enabled the time update notification"